packages/@atjson/renderer-react/src/index.ts

```diff
@@ -73,20 +72,32 @@ export default class ReactRenderer extends Renderer {
   ): Iterator<void, ReactElement | ReactElement[], ReactElement[]> {
     this.renderSubdocuments(annotation);
 
-    let AnnotationComponent =
-      this.componentLookup[annotation.type] ||
-      this.componentLookup[classify(annotation.type)];
+    const annotationChildren = yield;
+    const key = `${annotation.id}-${annotation.start}`;
 
-    if (AnnotationComponent) {
-      return React.createElement(
-        AnnotationComponent,
-        annotation.attributes,
-        ...(yield)
-      );
-    } else {
-      // console.warn(`No component found for "${node.type}"- content will be yielded`);
-      let components = yield;
-      return components;
-    }
+    return React.createElement(ReactRendererConsumer, {
+      key,
+      children: (componentMap: ComponentMap) => {
+        if (Object.keys(componentMap).length === 0) {
+          throw new Error(
```

Comment:
Worth making a custom error type here?

Reply:
I don't think that's necessary— this should be noisy 😬